### ISSUE TITLE
No support for editing NextResponse.next()'s value

### ISSUE THREAD

**Issue by Nitlix** [i18nexus/next-i18n-router#8]
I'm using the i18n router however I cannot edit any cookies or headers which get sent to the server after they have been passed through the middleware, which then doesn't allow middleware to communicate with the renderer

Here's an example:

```js
  import { NextResponse } from 'next/server'
  
  import themeSettings from "@/themes/settings"
  
  import { i18nRouter } from 'next-i18n-router';
  import i18nConfig from '../i18nConfig';
  
  const themeCookie = themeSettings.themeCookie;
  const lastThemeCookie = themeSettings.lastThemeCookie;
  
  export default function middleware(request) {
  
      // =================================
      // Create a response object
      // =================================
      const response = NextResponse.next({
          request: {
              // Apply old headers
              headers: request.headers,
              // Apply old cookies
              cookies: request.cookies
          }
      })
  
      // =================================
      // Define interaction functions
      // =================================
      function setCookie(name, value){
          request.cookies.set(name, value);
          response.cookies.set(name, value);
      }
  
      const getCookie = (name) => request.cookies.get(name);
  
      function setHeader(name, value){
          response.headers.set(name, value);
          request.headers.set(name, value);
      }
  
  
      //==================================================
      // Allow server components to access the URL
      //==================================================
      setHeader("x-url", request.url);
  
  
  
      //====================
      // Theme setting
      //====================
      let theme = getCookie(themeCookie);
      if (theme) {
          theme = theme.value.toLowerCase();
      }
  
      if (!themeSettings.allowed.includes(theme)) {
          theme = themeSettings.default;
          setCookie(themeCookie, themeSettings.default);
      }
  
      if (getCookie(lastThemeCookie)){
          setHeader(themeSettings.lastThemeHeader, getCookie(lastThemeCookie).value);
      }
  
      setHeader(themeSettings.themeHeader, theme);
  
  
      
      //===================================
      // Return response to the renderer
      // It'll handle it from here
      //===================================
      return i18nRouter(request, i18nConfig);
  }
  
  export const config = {
      // matcher solution for public, api, assets and _next exclusion
      matcher: "/((?!api|static|.*\\..*|_next).*)",
  };
```

Here, I'm trying to edit the response cookies so if they didn't exist before, the server can instantly access them after it has been passed through the middleware. However, the response object isn't parsed through the i18nRouter() function, meaning it can't reach the server, and only the library is able to create its own response object.

Could it please be added so that you can submit a custom response object that the router will then edit?

**Comment by Nitlix** [comment]
I've also submitted a PR with a possible fix to this issue, please take a look (;

**Comment by i18nexus** [comment]
Thanks for the feedback. Could you not just edit the response after running `i18nRouter`? For example:

```js
export default function middleware(request) {
   const response = i18nRouter(request, config);
   
   // custom logic
   
   return response;
}
```

Maybe I'm missing something.

**Comment by Nitlix** [comment]
Oh yeah, sorry my bad 😳

**Comment by Nitlix** [comment]
Didn't think of that, thanks for your feedback! I'll close my PR

**Comment by Nitlix** [comment]
Actually, I'll reopen the issue since it'd be great to implement something like this.

Because if every single library generated its own response object, then no information could actually be shared between them 😅...

**Comment by i18nexus** [comment]
Yeah... you're right 🙂

We'll review your PR and likely merge it soon.

**Comment by i18nexus** [comment]
Released in v3.2.0. Thanks for the PR!

**Comment by i18nexus** [comment]
Unfortunately we opted to get remove this functionality in v4.0.0. In order to simplify set up, we moved rewrites from `next.config.js` into the middleware in `i18nRouter`. We are unable (as far as we know) to run a rewrite on an existing response.

If you have an idea on how to add this functionality back while keeping rewrites in the `i18nRouter` function, we are happy to review any PR's 🙂 @Nitlix
